pkg/apis/openebs.io/scp/v1alpha1/affinity.go

```diff
@@ -0,0 +1,42 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Affinity is a group of affinity scheduling rules
+type Affinity struct {
+
+	// Describes volume affinity scheduling rules
+	// +optional
+	VolumeAffinity []VolumeAffinityTerm `json:"volumeAffinity,omitempty"`
+
+	// Describes volume anti-affinity scheduling rules
+	// +optional
+	VolumeAntiAffinity []VolumeAffinityTerm `json:"volumeAntiAffinity,omitempty"`
```

Comment:
`VolumeAntiAffinity` helps to place a volume in the different topology(storage cohort) where old volumes are present, if old volume not found the it can go to any cohort. Old volumes are identified using `LabelSelector`. Using this we can ensure that replicas of an application will have volume in different storage cohort topology . ie - if `app-0` and `app-1` replica of sts `app` is requesting for 2 volumes `data-0` and `data-1`, both the volumes will be scheduled in the different storage cohort topology.

Reply:
Please add this in the comment.

Reply:
also add these are ANDed.